### DISCUSSION TITLE
How to fix Inconsistent code blocks

### DISCUSSION THREAD

**Discussion by Momijie** [minad/org-modern#163] [Q&A]
Here is a screenshot of my literate config, where I have slimmed it down trying to find the issue.  
![Sat 02 Dec 2023-12 38 24 PM](https://github.com/minad/org-modern/assets/33807324/4dee5646-c4bd-4bdd-b97d-0a23b3b53237)
The issue mostly happens when unfolding a subtree. If I toggle 'org-modern-mode' it seems to temporarily fix the issue, but I don't see this as a fix. 

This is what I have currently for my org-modern config.

```
(defun satori/org/init-org-modern()
  ; putting this setq here for now.
  (setq org-startup-indented t)

  (use-package org-modern
    :hook
    (org-mode . org-modern-mode)
    (org-agenda-finalize . org-modern-agenda)))
(satori/org/init-org-modern)
```

Emacs 29.1
If it matters, I'm using elpaca to install the plugin.

**Comment by minad** [comment]
Does the problem also occur without `org-indent-mode`?

**Reply by Momijie** [reply]
It was appearing before I put  ```(setq org-startup-indented t)``` changing it to ```(setq org-startup-indented nil)``` doesn't seem to change anything. ```(org-indent-mode -1)``` Didn't change the situation either

**Reply by minad** [reply]
Are some of the `#+end_src` keywords indented in your source file? It looks like this, but anyway I cannot explain the issue. I've not seen it myself. Do you see it with emacs -Q? If yes, can you please paste a sample file for me to reproduce.

**Reply by Momijie** [reply]
https://github.com/Momijie/dotfiles/tree/Refactoring-Emacs/emacs/.config/emacs
Here's a link to my emacs directory in my dotfiles. Not much to look at in that branch but it should emulate the issue I'm having.

**Reply by minad** [reply]
For reproduction I need a minimal recipe based on emacs -Q.

**Reply by Momijie** [reply]
Oh sorry, do you mean like this?
```
(setq package-enable-at-startup nil)

(defvar elpaca-installer-version 0.6)
(defvar elpaca-directory (expand-file-name "elpaca/" user-emacs-directory))
(defvar elpaca-builds-directory (expand-file-name "builds/" elpaca-directory))
(defvar elpaca-repos-directory (expand-file-name "repos/" elpaca-directory))
(defvar elpaca-order '(elpaca :repo "https://github.com/progfolio/elpaca.git"
                              :ref nil
                              :files (:defaults "elpaca-test.el" (:exclude "extensions"))
                              :build (:not elpaca--activate-package)))
(let* ((repo  (expand-file-name "elpaca/" elpaca-repos-directory))
       (build (expand-file-name "elpaca/" elpaca-builds-directory))
       (order (cdr elpaca-order))
       (default-directory repo))
  (add-to-list 'load-path (if (file-exists-p build) build repo))
  (unless (file-exists-p repo)
    (make-directory repo t)
    (when (< emacs-major-version 28) (require 'subr-x))
    (condition-case-unless-debug err
        (if-let ((buffer (pop-to-buffer-same-window "*elpaca-bootstrap*"))
                 ((zerop (call-process "git" nil buffer t "clone"
                                       (plist-get order :repo) repo)))
                 ((zerop (call-process "git" nil buffer t "checkout"
                                       (or (plist-get order :ref) "--"))))
                 (emacs (concat invocation-directory invocation-name))
                 ((zerop (call-process emacs nil buffer nil "-Q" "-L" "." "--batch"
                                       "--eval" "(byte-recompile-directory \".\" 0 'force)")))
                 ((require 'elpaca))
                 ((elpaca-generate-autoloads "elpaca" repo)))
            (progn (message "%s" (buffer-string)) (kill-buffer buffer))
          (error "%s" (with-current-buffer buffer (buffer-string))))
      ((error) (warn "%s" err) (delete-directory repo 'recursive))))
  (unless (require 'elpaca-autoloads nil t)
    (require 'elpaca)
    (elpaca-generate-autoloads "elpaca" repo)
    (load "./elpaca-autoloads")))
(add-hook 'after-init-hook #'elpaca-process-queues)
(elpaca `(,@elpaca-order))

(setq org-startup-indented t)

(elpaca org-modern)

(defun satori/org/init-org-modern()
  ; putting this setq here for now.
  (setq org-startup-indented t)

  (use-package org-modern
    :hook
    (org-mode . org-modern-mode)
    (org-agenda-finalize . org-modern-agenda)))
(satori/org/init-org-modern)

; Open an org file
(find-file-other-window "~/.config/emacs/config.org")
```
and here's my org file I ran it on
```
#+Title: Satori's GNU Emacs Config
#+AUTHOR: Satori
#+DESCRIPTION: Satori's personal Emacs config.
#+STARTUP: content
#+OPTIONS: TOC:2

* Configuration
** Initial Setup
*** Source Scripts
#+begin_src emacs-lisp
(add-to-list 'load-path "~/.config/emacs/scripts/")
(require 'elpaca-setup)
(require 'buffer-move)
#+end_src
*** Config Functions
#+begin_src emacs-lisp
(defun disable-line-numbers ()
  (display-line-numbers-mode -1))

(defun disable-mode-line ()
  (setq mode-line-format nil))

(defun display-line-numbers-equalize ()
  "Equalize the width"
  (setq display-line-numbers-width (length (number-to-string (line-number-at-pos (point-max))))))
  #+end_src
*** Quiet Scratch Buffer
#+begin_src emacs-lisp
(setq initial-scratch-message "")
#+end_src
*** Misc
#+begin_src emacs-lisp
(recentf-mode 1)
#+end_src
** Visual Adjustments
*** Minimalde UI
#+begin_src emacs-lisp
(defun satori/visual/minimal-ui()
  (menu-bar-mode -1)
  (tool-bar-mode -1)
  (scroll-bar-mode -1))
;(satori/visual/minimal-ui)
#+end_src
*** Plugins
** Org Mode
*** Plugins
**** Org Modern
#+begin_src emacs-lisp
(defun satori/org/init-org-modern()
  ; putting this setq here for now.
  (setq org-startup-indented t)

  (use-package org-modern
    :hook
    (org-mode . org-modern-mode)
    (org-agenda-finalize . org-modern-agenda)))
(satori/org/init-org-modern)
#+end_src
**** Org Modern Indent
#+begin_src emacs-lisp
(defun satori/org/init-org-modern-indent()
  (use-package org-modern-indent 
    :elpaca (:host github :repo "jdtsmith/org-modern-indent")
    :config 
    (add-hook 'org-mode-hook #'org-modern-indent-mode 90)))
;(satori/org/init-org-modern-indent)
#+end_src
```
It's still happening here, so that's good.

**Reply by Momijie** [reply]
And an image for added context again.
![image](https://github.com/minad/org-modern/assets/33807324/403e3d62-c40b-4c32-9ca5-b30ef58d33c4)

**Reply by minad** [reply]
I don't see the issue with this recipe on Emacs 29.1. ~Do you use an older Emacs version?~ (EDIT: Nevermind you already mentioned 29.1 above.)

**Reply by Momijie** [reply]
That's weird, how did you unfold it?
For example move down to `***Source Scripts` and press TAB
If I remove  `#+STARTUP: content` and retry the script it all appears normally.
When I Change it to `#+STARTUP: fold` and use `(org-fold-show-all)` it appears normally.
While keeping it on _fold_ I went down to `***Source Scripts` pressed TAB and then ran `(org-fold-show-all)`, everything under `** Initial Setup` was broken but after that showed up correctly.

**Reply by minad** [reply]
Okay, I see the issue now. It is indeed related to folding and maybe triggered by the specific format of your Org file. 

(Btw. I suggest you don't use Org mode for your configuration if you only write source blocks under headlines. Then it is better to write a normal init.el and use outline-mode for folding.)

**Reply by Momijie** [reply]
Awesome!
If this is a bug, how can we go about fixing it and what part of the source is responsible for this piece of functionality?

Yeah, my org file is a bit of a mess right now, but I plan to do other things within it, I use babel tangle to extract the source code to its own .el file. Was loosely following the structure of a literate config from a few other people.

**Reply by minad** [reply]
This is likely not a bug in org-modern since it only happens when folding. As a first step one would have to understand why this happens, if some fontification rules are not executed or overridden or if the fontification breaks due to the folding.
